### PR TITLE
Wean Flink handlers off of global env interaction within `handle()`

### LINKED ISSUE #231: Migrate flink handlers off global env reads
Replace FLINK_ENV_NAME, FLINK_DATABASE_NAME, FLINK_COMPUTE_POOL_ID, FLINK_ENV_ID env reads in flink handlers with `runtime.config.getSoleConnection().flink` fields. 

For example ...

`list-flink-statements-handler.ts` reads `env.FLINK_COMPUTE_POOL_ID` as a Zod `.default()`
callback on the `computePoolId` argument, and reads `env.FLINK_ENV_ID` directly inside
`handle()` as a fallback when `environmentId` is not supplied.

Both map to fields on `conn.flink`: `compute_pool_id` and `environment_id`.

## What changes

Remove the Zod `.default()` callback for `computePoolId`. Read `conn.flink.compute_pool_id` and
`conn.flink.environment_id` inside `handle()` via `runtime.config.getSoleConnection()` as
fallbacks when the corresponding tool arguments are absent. Remove `import env from "@src/env.js"`.

## Tests

Add `handle()` tests covering:

- Explicit tool argument values take precedence over connection config
- Absent arguments fall back to connection config values
- Stub `ClientManager` with `createMockInstance(DefaultClientManager)`

### PR DESCRIPTION
**Pull request by jlrobins**
## Summary of Changes

- Migrate all 13 Flink handlers off `getEnsuredParam()` / `process.env` reads. Each handler
  now calls `this.getFlinkDirectConfig(runtime.config)` (new on `FlinkToolHandler`) to obtain
  the typed `FlinkDirectConfig` block, then reads `flink.organization_id`,
  `flink.environment_id`, and (where needed) `flink.compute_pool_id` as fallbacks when the
  corresponding tool argument is absent. `getEnsuredParam` is no longer mentioned anywhere
  under `handlers/flink/`.

- Add protected helper methods to `BaseToolHandler` and `FlinkToolHandler` that centralise the
  arg → config fallback pattern and document trimming behaviour in JSDoc:
  - `BaseToolHandler.resolveParam(arg, config, label)` — trims both sides, throws
    `"${label} is required"` if both absent.
  - `BaseToolHandler.resolveOptionalParam(arg, config)` — same trim logic, returns
    `undefined` instead of throwing.
  - `FlinkToolHandler.getFlinkDirectConfig(config)` — extracts and type-narrows the `flink`
    block from `getSoleConnection()`; throws `"Wacky -- "` if the block is absent, because the tool should not have been registered without it (see `FlinkToolHandler.getConnectionIds()`).
  - `FlinkToolHandler.resolveOrgAndEnvIds(flink, orgArg, envArg)` — wraps the two required
    ID resolves into one call, accepting `FlinkDirectConfig` directly.
  - `FlinkToolHandler.resolveComputePoolId` / `resolveOptionalComputePoolId` — required and
    optional variants for the compute pool ID.

- Introduce `FlinkCatalogToolHandler` (new intermediate base class) with a
  `resolveCatalogNameOrError()` method that returns a `CatalogResolution` discriminated union
  (`{ ok: true; name: string } | { ok: false; error: CallToolResult }`). All 5 catalog handlers
  now extend it and replace the previously duplicated 6-line error-path block with three lines:

  ```typescript
  const catalog = this.resolveCatalogNameOrError(catalogName, environment_id);
  if (!catalog.ok) return catalog.error;
  const catalog_name = catalog.name;
  ```

  The error path was previously untested across all five handlers; it is now covered once in
  `flink-catalog-tool-handler.test.ts`.

- Fix untrimmed config-value fallbacks caught during review:
  - `create-flink-statement-handler.ts`: `environment_name` / `database_name` now flow through
    `resolveOptionalParam()` so whitespace-only values fall back to config correctly.
  - `catalog-resolver.ts` `resolveDatabaseName()`: `conn.kafka.cluster_id` is now
    `.trim() || undefined` rather than returned raw.

- Remove dead "throw when X absent" test cases (3 inline `HandleCase` entries and 4
  `ZERO_ARG_OUTCOMES` entries) that required `connectionConfig: {}` to bypass Zod validation.
  Since `organization_id`, `environment_id`, and `compute_pool_id` are all required non-empty
  fields in `FlinkDirectConfig`, those throw paths are unreachable in production. The
  `"Wacky -- "` guard on `getFlinkDirectConfig` is tested once at the `FlinkToolHandler`
  level instead.

- Add `handle()` test suites to all 13 Flink handlers, covering config-fallback and
  explicit-arg cases per parameter. Additional targeted tests added during review:
  - Partial-override cases for `resolveOrgAndEnvIds` (org-only arg / env-only arg).
  - `detect-issues` explicit `computePoolId` verified against the telemetry POST body via
    `capturedCalls`.
  - `create-flink-statement` `capturedCalls` assertions verifying that
    `sql.current-catalog` / `sql.current-database` land correctly in POST `spec.properties`
    across all four combinations (config, explicit args, blank args, absent fields).
  - `resolveDatabaseName` whitespace-trim edge cases in `catalog-resolver.test.ts`.

## While here

- Update `ZERO_ARG_OUTCOMES` entries for the three catalog handlers (`list-flink-catalogs`,
  `list-flink-databases`, `list-flink-tables`) from `{ throws: "Organization ID is required" }`
  to `{ resolves: "..." }` — the conn-based fallback now succeeds against `allServicesRuntime`,
  so they reach the client layer instead of throwing.
- Add pointer to `BaseToolHandler` / `FlinkToolHandler` helper methods in
  `.claude/rules/tool-handlers.md` and `.github/copilot-instructions.md` so future contributors
  know to reach for them before writing inline fallback logic.
 - Removed the unused (and explicitily void'd) `databaseName` from list-tables-handler.ts.
- Now at **62%** line coverage.

## Size

| Bucket                                  | Added     | Deleted  | Total touched |
| --------------------------------------- | --------- | -------- | ------------- |
| `src/` non-test                         | +221      | −232     | **453**       |
| `src/` `.test.ts`                       | +1399     | −187     | **1586**      |
| Outside `src/` (rules, factories, etc.) | +30       | −1       | **31**        |
| **Overall**                             | **+1650** | **−420** | **2070**      |

Production code is net-negative (−11 lines) due to centralizing refactoring. 77% of all touched lines are test code, and these handlers needed (and continue to need) a ton.

## Relationships

- Closes #231.
- First step towards completing #230.

## Pull request checklist

Please check if your PR fulfills the following (if applicable):

#### Tests

- [x] Added new
- [x] Updated existing
- [ ] Deleted existing

#### Release notes

- [ ] Does anything in this PR need to be mentioned in the user-facing [CHANGELOG](https://github.com/confluentinc/mcp-confluent/blob/main/CHANGELOG.md)?

### CODE REVIEW

File: src/confluent/tools/handlers/flink/create-flink-statement-handler.ts [hunk 1]
```diff
@@ -76,21 +72,17 @@ export class CreateFlinkStatementHandler extends FlinkToolHandler {
...
+    const organization_id = organizationId || conn.flink?.organization_id;
+    if (!organization_id) throw new Error("Organization ID is required");
+    const environment_id = environmentId || conn.flink?.environment_id;
+    if (!environment_id) throw new Error("Environment ID is required");
+    const compute_pool_id = computePoolId || conn.flink?.compute_pool_id;
+    if (!compute_pool_id) throw new Error("Compute Pool ID is required");
+    const resolvedCatalogName =
+      catalogName ?? conn.flink?.environment_name ?? "";
+    const resolvedDatabaseName =
+      databaseName ?? conn.flink?.database_name ?? "";
```
[the remark this answers is not in the document]
Comment (jlrobins) [reply src/confluent/tools/handlers/flink/create-flink-statement-handler.ts]: Fixed and new test case added.

File: tests/stubs/handler.ts [hunk 2]
```diff
@@ -191,7 +260,7 @@ export async function assertHandleCase(options: {
     thrown = err;
   }
 
-  if (outcome === "TODO") {
+  if (outcome === "DISCOVER") {
     let discovered: string;
     if (thrown !== undefined) {
       discovered = `{ throws: ${JSON.stringify(classifyThrown(name, thrown))} }`;
```
[the remark this answers is not in the document]
Comment (jlrobins) [reply tests/stubs/handler.ts:315]: Thank you. The person responsible for the mistake has been sacked.

File: src/confluent/tools/handlers/flink/catalog/list-databases-handler.ts [hunk 3]
```diff
@@ -42,26 +41,27 @@ export class ListDatabasesHandler extends FlinkToolHandler {
...
-      "Compute Pool ID is required",
+    const compute_pool_id = this.resolveParam(
       computePoolId,
+      conn.flink?.compute_pool_id,
+      "Compute Pool ID",
     );
-    // Smart resolution: only accept env-* format, otherwise fall back to FLINK_ENV_ID
-    const catalog_name = resolveCatalogName(catalogName);
+    // Smart resolution: only accept env-* format, otherwise fall back to flink.environment_id from connection config
+    const catalog_name = resolveCatalogName(catalogName, conn);
```
[the remark this answers is not in the document]
Comment (jlrobins) [reply src/confluent/tools/handlers/flink/catalog/list-databases-handler.ts]: Fixed resolveCatalogName to use already-resolved environment_id, not raw conn

File: src/confluent/tools/handlers/flink/catalog/get-table-info-handler.ts [hunk 4]
```diff
@@ -65,31 +64,32 @@ export class GetTableInfoHandler extends FlinkToolHandler {
...
-      "Compute Pool ID is required",
+    const compute_pool_id = this.resolveParam(
       computePoolId,
+      conn.flink?.compute_pool_id,
+      "Compute Pool ID",
     );
-    // Smart resolution: only accept env-* format, otherwise fall back to FLINK_ENV_ID
-    const catalog_name = resolveCatalogName(catalogName);
+    // Smart resolution: only accept env-* format, otherwise fall back to flink.environment_id from connection config
+    const catalog_name = resolveCatalogName(catalogName, conn);
```
[the remark this answers is not in the document]
Comment (jlrobins) [reply src/confluent/tools/handlers/flink/catalog/get-table-info-handler.ts]: Fixed resolveCatalogName to use already-resolved environment_id, not raw conn

File: src/confluent/tools/handlers/flink/catalog/list-catalogs-handler.ts [hunk 5]
```diff
@@ -35,26 +34,27 @@ export class ListCatalogsHandler extends FlinkToolHandler {
...
-      "Compute Pool ID is required",
+    const compute_pool_id = this.resolveParam(
       computePoolId,
+      conn.flink?.compute_pool_id,
+      "Compute Pool ID",
     );
-    // Smart resolution: use FLINK_ENV_ID as the catalog name
-    const catalog_name = resolveCatalogName();
+    // Smart resolution: use flink.environment_id from connection config as the catalog name
+    const catalog_name = resolveCatalogName(undefined, conn);
```
[the remark this answers is not in the document]
Comment (jlrobins) [reply src/confluent/tools/handlers/flink/catalog/list-catalogs-handler.ts]: Fixed resolveCatalogName to use already-resolved environment_id, not raw conn

File: src/confluent/tools/handlers/flink/catalog/describe-table-handler.ts [hunk 6]
```diff
@@ -65,31 +64,32 @@ export class DescribeTableHandler extends FlinkToolHandler {
...
-      "Compute Pool ID is required",
+    const compute_pool_id = this.resolveParam(
       computePoolId,
+      conn.flink?.compute_pool_id,
+      "Compute Pool ID",
     );
-    // Smart resolution: only accept env-* format, otherwise fall back to FLINK_ENV_ID
-    const catalog_name = resolveCatalogName(catalogName);
+    // Smart resolution: only accept env-* format, otherwise fall back to flink.environment_id from connection config
+    const catalog_name = resolveCatalogName(catalogName, conn);
```
[the remark this answers is not in the document]
Comment (jlrobins) [reply src/confluent/tools/handlers/flink/catalog/describe-table-handler.ts]: Fixed resolveCatalogName to use already-resolved environment_id, not raw conn

File: src/confluent/tools/handlers/flink/catalog/catalog-resolver.ts [hunk 7]
```diff
@@ -11,19 +11,22 @@ import env from "@src/env.js";
...
+): string | undefined {
   // If provided value looks like an environment ID, use it
   if (catalogName && catalogName.startsWith("env-")) {
     return catalogName;
   }
 
-  // Fall back to FLINK_ENV_ID (the actual environment ID)
-  return env.FLINK_ENV_ID;
+  // Fall back to conn.flink.environment_id
+  return conn?.flink?.environment_id;
```
[the remark this answers is not in the document]
Comment (jlrobins) [reply src/confluent/tools/handlers/flink/catalog/catalog-resolver.ts]: Fixed resolveCatalogName to use already-resolved environment_id, not raw conn

File: src/confluent/tools/handlers/flink/catalog/list-tables-handler.ts [hunk 8]
```diff
@@ -46,40 +45,35 @@ export class ListTablesHandler extends FlinkToolHandler {
...
-      "Compute Pool ID is required",
+    const compute_pool_id = this.resolveParam(
       computePoolId,
+      conn.flink?.compute_pool_id,
+      "Compute Pool ID",
     );
-    // Smart resolution: only accept env-* format, otherwise fall back to FLINK_ENV_ID
-    const catalog_name = resolveCatalogName(catalogName);
+    // Smart resolution: only accept env-* format, otherwise fall back to flink.environment_id from connection config
+    const catalog_name = resolveCatalogName(catalogName, conn);
```
[the remark this answers is not in the document]
Comment (jlrobins) [reply src/confluent/tools/handlers/flink/catalog/list-tables-handler.ts]: Fixed resolveCatalogName to use already-resolved environment_id, not raw conn

File: src/confluent/tools/base-tools.ts [hunk 9]
```diff
@@ -68,6 +68,22 @@ export abstract class BaseToolHandler implements ToolHandler {
...
+   * a connection-config value. Throws if neither is present.
+   * `label` is the human-readable field name (e.g. `"Organization ID"`);
+   * the thrown message is `"${label} is required"`.
+   */
+  protected resolveParam(
+    argValue: string | undefined,
+    configValue: string | undefined,
+    label: string,
+  ): string {
+    const resolved = argValue || configValue;
```
[the remark this answers is not in the document]
Comment (jlrobins) [reply src/confluent/tools/base-tools.ts]: Fixed, tested.

File: src/confluent/tools/handlers/flink/list-flink-statements-handler.ts [hunk 10]
```diff
@@ -20,7 +18,6 @@ const listFlinkStatementsArguments = z.object({
   computePoolId: z
     .string()
     .optional()
-    .default(() => env.FLINK_COMPUTE_POOL_ID ?? "")
     .describe("Filter the results by exact match for compute_pool."),
```
[the remark this answers is not in the document]
Comment (jlrobins) [reply src/confluent/tools/handlers/flink/list-flink-statements-handler.ts:21]: Fixed:

All three fields flow through resolveOrgAndEnvIds / resolveOptionalComputePoolId, which call resolveParam / resolveOptionalParam — both do argValue?.trim() before the fallback check. A whitespace-only "   " becomes "" there, which is falsy, so the config value is used instead. The trimming happens a layer down rather than at the Zod boundary, but the behavior is identical

File: src/confluent/tools/handlers/kafka/get-topic-config.ts [hunk 11]
```diff
@@ -38,10 +37,11 @@ export class GetTopicConfigHandler extends BaseToolHandler {
...
       getTopicConfigArguments.parse(toolArguments);
-    const kafka_cluster_id = getEnsuredParam(
-      "KAFKA_CLUSTER_ID",
-      "Kafka Cluster ID is required",
+    const conn = runtime.config.getSoleConnection();
+    const kafka_cluster_id = this.resolveParam(
       clusterId,
+      conn.kafka?.cluster_id,
+      "Kafka Cluster ID",
     );
```
[the remark this answers is not in the document]
Comment (jlrobins) [reply src/confluent/tools/handlers/kafka/get-topic-config.ts:45]: Fixed.

File: src/confluent/tools/handlers/kafka/alter-topic-config.ts [hunk 12]
```diff
@@ -46,10 +45,11 @@ export class AlterTopicConfigHandler extends BaseToolHandler {
...
       alterTopicConfigArguments.parse(toolArguments);
-    const kafka_cluster_id = getEnsuredParam(
-      "KAFKA_CLUSTER_ID",
-      "Kafka Cluster ID is required",
+    const conn = runtime.config.getSoleConnection();
+    const kafka_cluster_id = this.resolveParam(
       clusterId,
+      conn.kafka?.cluster_id,
+      "Kafka Cluster ID",
     );
```
[the remark this answers is not in the document]
Comment (jlrobins) [reply src/confluent/tools/handlers/kafka/alter-topic-config.ts:53]: Fixed.

File: src/confluent/tools/handlers/flink/list-flink-statements-handler.test.ts [hunk 13]
```diff
@@ -30,26 +17,21 @@ describe("list-flink-statements-handler.ts", () => {
...
-          outcome: { throws: "Organization ID is required" },
+          outcome: { resolves: "{}" },
         },
         {
-          label: "resolves when required IDs are supplied as explicit args",
+          label: "resolve when required IDs are supplied as explicit args",
           args: EXPLICIT_IDS,
           outcome: { resolves: "{}" },
         },
         {
```
[the remark this answers is not in the document]
Comment (jlrobins) [reply src/confluent/tools/handlers/flink/list-flink-statements-handler.test.ts:33]: The comment is stale. The "Wacky -- " path from getFlinkDirectConfig() is already tested — it lives in flink-tool-handler.test.ts rather than in each individual handler test. That was intentional.

File: src/confluent/tools/handlers/flink/catalog/list-catalogs-handler.test.ts [hunk 14]
```diff
@@ -0,0 +1,91 @@
...
+          label: "use explicit org/env/compute args over config",
+          args: {
+            organizationId: "org-from-args",
+            environmentId: "env-from-args",
+            computePoolId: "lfcp-from-args",
+          },
+          outcome: { resolves: "Catalogs" },
+          responseData: SQL_RESPONSE,
+        },
+      ];
```
[the remark this answers is not in the document]
Comment (jlrobins) [reply src/confluent/tools/handlers/flink/catalog/list-catalogs-handler.test.ts:38]: Same as above -- stale.

File: src/confluent/tools/handlers/flink/catalog/list-tables-handler.test.ts [hunk 15]
```diff
@@ -0,0 +1,105 @@
...
+          label: "use explicit org/env/compute args over config",
+          args: {
+            organizationId: "org-from-args",
+            environmentId: "env-from-args",
+            computePoolId: "lfcp-from-args",
+          },
+          outcome: { resolves: "Tables in catalog" },
+          responseData: SQL_RESPONSE,
+        },
+      ];
```
[the remark this answers is not in the document]
Comment (jlrobins) [reply src/confluent/tools/handlers/flink/catalog/list-tables-handler.test.ts:40]: Same, stale.

File: src/confluent/tools/handlers/flink/create-flink-statement-handler.test.ts [hunk 16]
```diff
@@ -1,75 +1,222 @@
...
+const FLINK_CONN = {
+  flink: {
+    ...BASE_FLINK_CONN.flink,
+    environment_name: "env-name-from-config",
+    database_name: "db-name-from-config",
+  },
+};
+
+// Flink config without the optional environment_name / database_name fields,
+// used to exercise the ?? "" fallback in resolvedCatalogName / resolvedDatabaseName.
```
[the remark this answers is not in the document]
Comment (jlrobins) [reply src/confluent/tools/handlers/flink/create-flink-statement-handler.test.ts]: Fixed.

File: src/confluent/tools/handlers/flink/catalog/list-databases-handler.test.ts [hunk 17]
```diff
@@ -0,0 +1,103 @@
...
+          label: "use explicit org/env/compute args over config",
+          args: {
+            organizationId: "org-from-args",
+            environmentId: "env-from-args",
+            computePoolId: "lfcp-from-args",
+          },
+          outcome: { resolves: "Databases" },
+          responseData: SQL_RESPONSE,
+        },
+      ];
```
[the remark this answers is not in the document]
Comment (jlrobins) [reply src/confluent/tools/handlers/flink/catalog/list-databases-handler.test.ts:38]: Same issue. See above.

File: src/confluent/tools/handlers/flink/catalog/catalog-resolver.ts [hunk 18]
```diff
@@ -1,29 +1,32 @@
...
+  fallbackEnvId?: string,
+): string | undefined {
   if (catalogName && catalogName.startsWith("env-")) {
     return catalogName;
   }
 
-  // Fall back to FLINK_ENV_ID (the actual environment ID)
-  return env.FLINK_ENV_ID;
+  return fallbackEnvId;
 }
```
[the remark this answers is not in the document]
Comment (jlrobins) [reply src/confluent/tools/handlers/flink/catalog/catalog-resolver.ts:36]: Fixed.

File: src/confluent/tools/handlers/flink/catalog/flink-catalog-tool-handler.ts [hunk 19]
```diff
@@ -0,0 +1,26 @@
...
+  protected resolveCatalogNameOrError(
+    catalogName: string | undefined,
+    environmentId: string,
+  ): CatalogResolution {
+    const name = resolveCatalogName(catalogName, environmentId);
+    if (!name)
+      return {
+        ok: false,
+        error: this.createResponse(
+          "Catalog name could not be resolved. Set flink.environment_id in config or provide a valid environment ID (env-xxxxx).",
```
[the remark this answers is not in the document]
Comment (jlrobins) [reply src/confluent/tools/handlers/flink/catalog/flink-catalog-tool-handler.ts]: Fixed.

File: src/confluent/tools/handlers/flink/catalog/list-tables-handler.ts [hunk 20]
```diff
@@ -40,46 +38,27 @@ const listTablesArguments = z.object({
...
     const clientManager = runtime.clientManager;
-    const {
-      organizationId,
-      environmentId,
-      computePoolId,
-      catalogName,
-      databaseName,
-    } = listTablesArguments.parse(toolArguments);
+    const { organizationId, environmentId, computePoolId, catalogName } =
+      listTablesArguments.parse(toolArguments);
```
[the remark this answers is not in the document]
Comment (jlrobins) [reply src/confluent/tools/handlers/flink/catalog/list-tables-handler.ts:41]: Removed the field. Sigh.

File: src/confluent/tools/handlers/flink/catalog/catalog-resolver.ts (hunk 18 above)
[the remark this answers is not in the document]
Comment (jlrobins) [reply src/confluent/tools/handlers/flink/catalog/catalog-resolver.ts:36]: Fixed.

File: src/confluent/tools/handlers/flink/catalog/flink-catalog-tool-handler.ts [hunk 21]
```diff
@@ -0,0 +1,26 @@
...
+  protected resolveCatalogNameOrError(
+    catalogName: string | undefined,
+    environmentId: string,
+  ): CatalogResolution {
+    const name = resolveCatalogName(catalogName, environmentId);
+    if (!name)
+      return {
+        ok: false,
+        error: this.createResponse(
+          "Catalog name could not be resolved. Pass catalogName (env-xxxxx) explicitly, or set flink.environment_id in config.",
```
[the remark this answers is not in the document]
Comment (jlrobins) [reply src/confluent/tools/handlers/flink/catalog/flink-catalog-tool-handler.ts]: Fixed.

File: src/confluent/tools/handlers/flink/create-flink-statement-handler.test.ts [hunk 22]
```diff
@@ -1,75 +1,223 @@
...
+        expect(capturedCalls[0]!.args).toMatchObject({
+          body: expect.objectContaining({
+            spec: expect.objectContaining({
+              properties: {
+                "sql.current-catalog": FLINK_CONN.flink.environment_name,
+                "sql.current-database": FLINK_CONN.flink.database_name,
+              },
+            }),
+          }),
+        });
```
Comment (shouples) [inline src/confluent/tools/handlers/flink/create-flink-statement-handler.test.ts:146]: More for https://github.com/confluentinc/mcp-confluent/issues/323

File: src/confluent/tools/handlers/flink/catalog/describe-table-handler.ts [hunk 23]
```diff
@@ -65,31 +63,19 @@ export class DescribeTableHandler extends FlinkToolHandler {
...
-        true,
-      );
-    }
+    const compute_pool_id = this.resolveComputePoolId(flink, computePoolId);
+    const catalog = this.resolveCatalogNameOrError(catalogName, environment_id);
+    if (!catalog.ok) return catalog.error;
+    const catalog_name = catalog.name;
     // Database name is optional - if provided, resolve it to friendly SCHEMA_NAME
-    const database_input = resolveDatabaseName(databaseName);
+    const database_input = resolveDatabaseName(databaseName, conn);
```
Comment (shouples) [inline src/confluent/tools/handlers/flink/catalog/describe-table-handler.ts:78]: If the `kafka` config uses different `auth` than the `flink` block, will reusing the cluster ID as a database ID still work? Or do we expect it to fall through if the Flink requests can't communicate with the set database (due to mismatched provider/region or otherwise)?

**Review by shouples — APPROVED** [review]
Solid progress away from `env`, plus I appreciate some of the extra cleanup-related changes as well 👏

File: src/confluent/tools/handlers/flink/catalog/describe-table-handler.ts (hunk 23 above)
Comment (jlrobins) [reply src/confluent/tools/handlers/flink/catalog/describe-table-handler.ts:78]: I .. imagine the original author expected/assumed the Kafka cluster config and the Flink db config to be referencing ... the same Kafka cluster in both Kafka and Flink Database duality here? In that case, it would work. If the configured Kafka cluster is in a different cloud/region from the Flink config, then probabyl a hard fail. If is a different cluster, but the same cloud/region, then we'll get results, just not what the user might expect?

This changeset's goal was to (try to _only_ ) respell how we get at any preexisting assumptions -- no longer off of the env vars directly.

File: src/confluent/tools/handlers/flink/create-flink-statement-handler.test.ts (hunk 22 above)
Comment (jlrobins) [reply src/confluent/tools/handlers/flink/create-flink-statement-handler.test.ts:146]: There will be more coming. #323 should just blindly target any `capturedCalls` use.

Comment (shouples) [reply src/confluent/tools/handlers/flink/create-flink-statement-handler.test.ts:146]: Yeah... I'll try to prep the 323 branch today to get ahead of future offenders 😅

**Review by shouples — APPROVED** [review]
